### PR TITLE
refactor: rename `Constants.OPCode` to `Opcode`, moved objects to enums

### PR DESCRIPTION
**Pull request by kyranet**
**Please describe the changes this PR makes and why it should be merged:**

Renamed `Constants.OPCode` for consistency with recent changes, also moved enum-like objects to the enums file and added tests to ensure they don't receive properties that shouldn't exist.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: typings/enums.d.ts [hunk 1]
```diff
@@ -152,3 +152,140 @@ export enum WebhookTypes {
...
+
+export enum ConstantsShardEvents {
+  CLOSE = 'close',
+  DESTROYED = 'destroyed',
+  INVALID_SESSION = 'invalidSession',
+  READY = 'ready',
+  RESUMED = 'resumed',
+}
+
+export enum ConstantsColors {
```
Comment (SpaceEEC) [inline typings/enums.d.ts:231]: This is not an enum, but TS allows compiling something like `Colors[Colors.RED]`, yielding a `string`, however in runtime it's actually `undefined`.
We could make it an enum however, `createEnum` could be modified to accept objects too.

File: typings/enums.d.ts [hunk 2]
```diff
@@ -152,3 +152,140 @@ export enum WebhookTypes {
...
+  DARKER_GREY = 0x7f8c8d,
+  LIGHT_GREY = 0xbcc0c0,
+  DARK_NAVY = 0x2c3e50,
+  BLURPLE = 0x5865f2,
+  GREYPLE = 0x99aab5,
+  DARK_BUT_NOT_BLACK = 0x2c2f33,
+  NOT_QUITE_BLACK = 0x23272a,
+}
+
+export enum ConstantsStatus {
```
Comment (SpaceEEC) [inline typings/enums.d.ts:264]: This one too.

File: typings/enums.d.ts [hunk 3]
```diff
@@ -152,3 +152,140 @@ export enum WebhookTypes {
...
+  READY = 0,
+  CONNECTING = 1,
+  RECONNECTING = 2,
+  IDLE = 3,
+  NEARLY = 4,
+  DISCONNECTED = 5,
+}
+
+export enum ConstantsOpcodes {
```
Comment (SpaceEEC) [inline typings/enums.d.ts:273]: This one too.

File: typings/enums.d.ts [hunk 4]
```diff
@@ -152,3 +152,140 @@ export enum WebhookTypes {
...
+  VOICE_GUILD_PING = 5,
+  RESUME = 6,
+  RECONNECT = 7,
+  REQUEST_GUILD_MEMBERS = 8,
+  INVALID_SESSION = 9,
+  HELLO = 10,
+  HEARTBEAT_ACK = 11,
+}
+
+export enum ConstantsClientApplicationAssetTypes {
```
Comment (SpaceEEC) [inline typings/enums.d.ts:288]: This one too.

**Review by SpaceEEC — CHANGES_REQUESTED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**